src/com/lftechnology/java/training/niraj/ems/services/implementations/UserServiceImpl.java

```diff
@@ -0,0 +1,507 @@
+        }
+        return employee;
+    }
+
+    /**
+     * Get department by id
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param id
+     * @return
+     */
+    private String getDepartmentById(int id) {
+        String department;
+        switch (id) {
+        case 2:
+            department = "ROR";
+            break;
+        case 3:
+            department = "PHP";
+            break;
+        case 4:
+            department = "Devops";
+            break;
+        case 5:
+            department = "Android";
+            break;
+        case 6:
+            department = "iOS";
+            break;
+        case 7:
+            department = "Administration";
+            break;
+        case 8:
+            department = "Operations";
+            break;
+        default:
+            department = "Java";
+            break;
+        }
+        return department;
+    }
+
+    /**
+     * Parses user information
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param attributes
+     * @param userInfo
+     */
+    private void parseUserInfo(List<String> attributes, Map<String, String> userInfo) {
+        for (String attribute : attributes) {
+            if (!userInfo.containsKey(attribute)) {
+                userInfo.put(attribute, null);
+            }
+        }
+        if (userInfo.get("status") == null) {
+            userInfo.put("status", Status.ACTIVE.getStatus());
+        }
+        if (userInfo.get("isTerminated") == null) {
+            userInfo.put("isTerminated", Status.INACTIVE.getStatus());
+        }
+    }
+
+    /**
+     * Sets attributes
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * 
+     * @param employee
+     * @param userInfo
+     * @return
+     */
+    public Employee setAttributes(Employee employee, Map<String, String> userInfo) {
+        employeeDao.setEmployeeObject(userInfo, employee);
+        return employee;
+
+    }
+
+    /**
+     * Searches user
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param scanner
+     * @param console
+     * @return
+     * @throws SQLException
+     */
+    public String searchUser(Scanner scanner, Console console) throws SQLException {
+        Map<String, String> searchCondition = new LinkedHashMap<String, String>();
+        String username = userInput.getInput(scanner, console, Constants.SEARCH_BY_USERNAME);
+        searchCondition.put("username", username);
+        searchCondition.put("fullname", username);
+        searchCondition.put("department", username);
+        searchCondition.put("address", username);
+        searchCondition.put(Constants.OPERATOR, "AND");
+        searchCondition.put("status", Status.ACTIVE.getStatus());
+        List<Employee> searchResults = employeeDao.findAll(searchCondition, Operators.OR);
+        if (searchResults.size() == 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1155)